### PR TITLE
[shopsys] moved setting common entity data to the method

### LINKED ISSUE #1746: DRY in entities (edit vs construct)
### What is happening
In many entities, you can find a nice example of DRY violation. Constructor and `edit` method often contain duplicated lines for setting entity properties from the class data. This might lead to mistakes and it also hardens extending.

It is easy to fix this problem - the common properties settings can be moved into a separate method.

### Expected result
DRY violations in entities are reduced using a protected method (named eg. `fillCommonProperties`)

### PR DESCRIPTION
**Pull request by s3tezsky**
| Q             | A
| ------------- | ---
|Description, reason for the PR| We have many duplicated logic in entities. This PR extracts common parts into separate methodd.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| closes #1746  
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: upgrade/UPGRADE-v9.1.0-dev.md [hunk 1]
```diff
@@ -39,3 +39,6 @@ There you can find links to upgrade notes for other versions too.
         - `Shopsys\FrameworkBundle\Model\Product\Product::$recalculateAvailability`
         - `Shopsys\FrameworkBundle\Model\Product\Product::$recalculatePrice`
         - `Shopsys\FrameworkBundle\Model\Product\Product::$recalculateVisibility`
+
+- moved setting of common entity data to new method ([#1976](https://github.com/shopsys/shopsys/pull/1976))
+    - see #project-base-diff1 and #project-base-diff2 to update your project
```
[the remark this answers is not in the document]
Comment (s3tezsky) [reply upgrade/UPGRADE-v9.1.0-dev.md:44]: I just copy&pasted the line from above without noticing that 👍

File: packages/framework/src/Model/Category/Category.php [hunk 2]
```diff
@@ -116,6 +115,15 @@ public function edit(CategoryData $categoryData)
         $this->setDomains($categoryData);
     }
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Category\CategoryData $categoryData
+     */
+    protected function setData(CategoryData $categoryData): void
```
[the remark this answers is not in the document]
Comment (s3tezsky) [reply packages/framework/src/Model/Category/Category.php:121]: Nice catch!

File: packages/framework/src/Model/Customer/User/CustomerUser.php [hunk 3]
```diff
@@ -137,27 +137,31 @@ class CustomerUser implements UserInterface, TimelimitLoginInterface, Serializab
...
-        $this->lastName = $customerUserData->lastName;
-        if ($customerUserData->createdAt !== null) {
-            $this->createdAt = $customerUserData->createdAt;
-        } else {
-            $this->createdAt = new \DateTime();
-        }
         $this->domainId = $customerUserData->domainId;
-        $this->pricingGroup = $customerUserData->pricingGroup;
-        $this->telephone = $customerUserData->telephone;
+        $this->setData($customerUserData);
```
[the remark this answers is not in the document]
Comment (s3tezsky) [reply packages/framework/src/Model/Customer/User/CustomerUser.php]: moved in aall entities

**Comment by s3tezsky** [comment]
> you have not updated OrderStatus.

Fixed

---

> I was thinking about removing __construct and edit methods from project-base extended entities, which now have setData method - what do you think? If you do not agree, you should at least add to each entity override of setData method, so it is obvious how to use it.

I have overriden `setData()` in all affected entities.

---

> Cookbook create-advanced-grid.md should be updated so entity Salesman uses __construct, edit and setData methods.

> Cookbook adding-new-attribute-to-an-entity.md should be updated to set extId via setData method.

Fixed

---

> Cookbook extending-product-list.md should be updated to use setData method for brandName.

Not fixed. The `brandName` assigning is not  in entity.

**Comment by TomasLudvik** [comment]
In `adding-new-attribute-to-an-entity.md` you still have to update first part on line 48

**Pull request merged**